airflow/plugins/utils.py

```diff
@@ -81,7 +153,369 @@ def get_successfully_downloaded_feeds(execution_date):
+    @classmethod
+    def get_latest(cls) -> "AirtableGTFSDataExtract":
+        # TODO: this concatenation should live on the abstract base class probably
+        latest = get_latest_file(cls.bucket_table(), keys=cls.partition_names)
+
+        logging.info(
+            f"identified {latest.name} as the most recent extract of gtfs datasets"
+        )
+
+        with get_fs().open(latest.name, "rb") as f:
+            content = gzip.decompress(f.read()).decode()
+
+        records = [
+            AirtableGTFSDataRecord(**json.loads(row)) for row in content.splitlines()
+        ]
+
+        return AirtableGTFSDataExtract(
+            records=records,
+            filename=latest.filename,
+            dt=pendulum.parse(latest.partition["dt"], exact=True),
+            time=pendulum.parse(latest.partition["time"], exact=True),
+        )
+
+
+class GTFSFeedExtractInfo(PartitionedGCSArtifact):
+    # TODO: this should check whether the bucket exists https://stackoverflow.com/a/65628273
+    # TODO: this should be named `gtfs-raw` _or_ we make it dynamic
+    bucket: ClassVar[str] = prefix_bucket("gs://gtfs-schedule-raw")
+    partition_names: ClassVar[List[str]] = ["dt", "base64_url", "time"]
+    config: AirtableGTFSDataRecord
+    response_code: int
+    response_headers: Optional[Dict[str, str]]
+    download_time: pendulum.DateTime
+
+    @property
+    def table(self) -> GTFSFeedType:
+        return self.config.data
+
+    @property
+    def dt(self) -> pendulum.Date:
+        return self.download_time.date()
+
+    @property
+    def base64_url(self) -> str:
+        return self.config.base64_encoded_url
+
+    @property
+    def time(self) -> pendulum.Time:
+        return self.download_time.time()
+
+
+class AirtableGTFSDataRecordProcessingOutcome(ProcessingOutcome):
+    input_type: ClassVar[typing.Type[PartitionedGCSArtifact]] = GTFSFeedExtractInfo
+    extract: Optional[GTFSFeedExtractInfo]
+
+
+# class GTFSFeedExtractOutcome(PartitionedGCSArtifact):
+#     extract: Optional[GTFSFeedExtract]
+#     success: bool
+#     exception: Optional[Exception]
+#     body: Optional[str]
+#
+#     class Config:
+#         arbitrary_types_allowed = True
+#         json_encoders = {Exception: lambda e: str(e)}
+#
+#     @property
+#     def table(self) -> str:
+#         return f"{self.extract.feed.type}_outcomes"
+#
+#     @property
+#     def bucket(self) -> str:
+#         pass
+
+
+if __name__ == "__main__":
+    # records = AirtableGTFSDataExtract.get_latest().records
+    # extract = GTFSFeedExtract(
+    #     filename="whatever",
+    #     config=records[0],
+    #     response_code=200,
+    #     response_headers={},
+    #     download_time=pendulum.now(),
+    # )
+    # extract.save_content(get_fs(), b"")
+    # print(
+    #     AirtableGTFSDataRecordProcessingOutcome(
+    #         success=True,
+    #         extract=extract,
+    #     ).json(indent=4)
+    # )
+    objs = fetch_all_in_partition(
```

Comment:
do we want to leave this hard coded in here? maybe this should be more clearly labeled as like a test date or something?

Reply:
Yes I can add a comment.